Libraries/NavigationExperimental/NavigationTypeDefinition.js

```diff
@@ -23,7 +23,8 @@ export type NavigationGestureDirection = 'horizontal' | 'vertical';
 
 export type NavigationRoute = {
   key: string,
-  title?: string
+  title?: string,
+  direction?: NavigationGestureDirection,
```

Comment:
@hedgerwang definitely a fair point - I'm already adding some props to pass to components in the route object and I could definitely see it growing in size to become unwieldy.

I haven't been following `NavigationExperimental` super closely so I'm not exactly sure the intent of the API design, but in this case I think it might make sense to add a few properties to the route if it is tightly coupled to a `NavigationCard`, like [`cardStyle`](https://github.com/facebook/react-native/blob/master/Libraries/CustomComponents/NavigationExperimental/NavigationCardStack.js#L65) and `direction`. Since the `NavigationCardStack` looks at these properties to render, I'm not sure that hiding it under another nested object would be clear.

Basically, if we're able to set some defaults for the whole card stack, it would also be nice to override those for specific cards - if there's a better place to do that then the route object then I'd definitely be interested in that, but from my quick reading of the code this seemed like the easiest place to make that happen.